Aplib.Core/ILoggable.cs

```diff
@@ -0,0 +1,79 @@
+using System.Collections.Generic;
+
+namespace Aplib.Core
+{
+    /// <summary>
+    /// An interface that allows defining the structure loggable objects through a tree.
+    /// </summary>
+    public interface ILoggable : IDocumented
+    {
+        /// <summary>
+        /// Gets the children of the loggable object.
+        /// </summary>
+        /// <returns>The children of the loggable object.</returns>
+        public IEnumerable<ILoggable> GetChildren();
```

Comment:
Since it is an interface; GetChildren is a name that is prevalent in tree-like classes, so it could easily clash with other class members when implemented.
```suggestion
        public IEnumerable<ILoggable> GetLogChildren();
```